### PR TITLE
Improved back-to-top Button

### LINKED ISSUE #3387: Improving the back-to-top button
### Please describe the problem (or idea)
The Back-to-top button is present at the end of the webpage and is static there. Instead it can be present at the bottom-right corner of the screen all the time whenever someone scrolls down the page to make the user expirience better.
> What happened just before the problem occurred? Or what problem could this idea solve?

> What did you expect to see that you didn't?

There should have been a back-to-top button all the time at the bottom right corner whenever someone scrolls down the page.

### Please show us where to look

https://publiclab.org/dashboard

### What's your PublicLab.org username?
rbhatia46
> This can help us diagnose the issue: 

### Browser, version, and operating system
Chrome Version 69.0.3497.100, Windows 10
> Many bugs are related to these -- please help us track it down and reproduce what you're seeing!

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site. 

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by rbhatia46**
Fixes #3387 
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] PR body includes `fixes #0000`-style reference to original issue #
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by rbhatia46** [comment]
@publiclab/reviewers Please review.

**Comment by SidharthBansal** [comment]
Can you share a gif of proper working ?

**Comment by jywarren** [comment]
Also, I worry it may overlap other content... What happens on the /post
page? Thanks!

**Comment by rbhatia46** [comment]
@SidharthBansal @jywarren It would be placed 20px from the bottom-right corner and I guess that much space is vacant on every single route so I think this will not overlap with the content and it will be visible only when we scroll at least 50px on the page. Do let me know if these values need to be changed. 
Thanks!

**Comment by jywarren** [comment]
This sounds good; can you share some screenshots? What if we set the scroll threshold quite a bit more, like 1000px, so that it really only appears when the page gets quite long?

**Comment by rbhatia46** [comment]
@jywarren I am facing some issues running the code locally on Windows 10 with some dependencies(libv8 and a couple more). But I guess 1000px will be a bit too much. I was looking to make it visible as soon as the user scrolls the page a bit, it won't cover the entire page and just float around at the bottom-right corner. I was trying to achieve something like [this ](https://codepen.io/matthewcain/pen/ZepbeR). I guess 500px will be a good value, what do you say, please let me know so that I can change accordingly.

Thanks!

**Comment by jywarren** [comment]
Hi, just checking if you've gotten stuck on this at all, or if I could help in any way? Thanks!

**Comment by rbhatia46** [comment]
Hi @jywarren, I was busy with exams this week and now I am done with them now. Was having some issues installing plots2 on Windows 10(some gem installation issue). Will try on Ubuntu  using a VM.

**Comment by SidharthBansal** [comment]
OK no problem. If you still face any difficulty in installing software
please contact at gitter chat.

Best of luck for exam results.

**Comment by SidharthBansal** [comment]
Are you stuck on this? We can help you.

File: app/assets/javascripts/header_footer.js [hunk 1]
```diff
@@ -1,11 +1,19 @@
 jQuery(document).ready(function() {	 
-	var duration = 300;
+	
+
+	$(window).scroll(function() {
+    if ($(this).scrollTop() >= 50) {        // If page is scrolled more than 50px
+        $('.back-to-top').fadeIn(200);    // Fade in the button
```
Comment (HarshKhandeparkar) [inline app/assets/javascripts/header_footer.js]: Instead of using fadeIn, you should probably animate the opacity to something like 0.7

Comment (HarshKhandeparkar) [reply app/assets/javascripts/header_footer.js]: You can also remove the event listener after the button fades in and add after the scroll to top is complete.

**Comment by SidharthBansal** [comment]
As the person is inactive for more than a month, I am closing the PR. In case you want to push changes please feel free to open a new PR OR reopen this PR and add additional changes to it.
Thanks for contributing on Public Lab
cc @jywarren @gauravano

**Pull request closed without merging**